vm-disk-performance-meter/azuredeploy.json

```diff
@@ -0,0 +1,323 @@
+      "metadata": {
+        "description": "Size of the VM that runs the test."
+      }
+    },
+    "dataDiskStorageAccountType": {
+      "type": "string",
+      "defaultValue": "Standard_LRS",
+      "allowedValues": [
+        "Standard_LRS",
+        "Standard_GRS",
+        "Standard_ZRS",
+        "Premium_LRS"
+      ],
+      "metadata": {
+        "description": "Storage Account type for the test disk"
+      }
+    },
+    "dataDiskHostCaching": {
+      "type": "string",
+      "defaultValue": "ReadWrite",
+      "allowedValues": [
+        "",
+        "ReadOnly",
+        "ReadWrite"
+      ],
+      "metadata": {
+        "description": "Storage Account type for the test disk"
+      }
+    },
+    "testType": {
+      "type": "string",
+      "allowedValues": [
+        "read",
+        "write",
+        "randread",
+        "randwrite",
+        "randtrim",
+        "rw",
+        "readwrite",
+        "randrw"
+      ],
+      "defaultValue": "randrw",
+      "metadata": {
+        "description": "Disk test type to run. (rand: random, sequential otherwise; rw - read/write)"
+      }
+    },
+    "testSize": {
+      "type": "string",
+      "allowedValues": [
+        "128m",
+        "256m",
+        "512m",
+        "1g",
+        "2g",
+        "10g",
+        "30g"
+      ],
+      "defaultValue": "1g",
+      "metadata": {
+        "description": "Size of the file to test."
+      }
+    },
+    "secondsToRunTest": {
+      "type": "int",
+      "defaultValue": 30,
+      "metadata": {
+        "description": "Number of seconds for the test to run."
+      }
+    },
+      "adminUsername": {
+        "type": "string",
+        "metadata": {
+          "description": "Username for the test VMs."
+        }
+      },
+      "adminPassword": {
+        "type": "securestring",
+        "metadata": {
+          "description": "Password for the test VMs."
+        }
+      }
+    },
+  "variables": {
+    "osVersion": "14.04.5-LTS",
+    "imagePublisher": "Canonical",
+    "imageOffer": "UbuntuServer",
+    "OSDiskName": "osdiskforwindowssimple",
+    "nicName": "testVMNic",
+    "addressPrefix": "10.0.0.0/16",
+    "subnetName": "Subnet",
+    "subnetPrefix": "10.0.0.0/24",
+    "storageAccountType": "Standard_LRS",
+    "publicIPAddressName": "publicIP",
+    "publicIPAddressType": "Static",
+    "vmStorageAccountContainerName": "vhds",
+    "virtualNetworkName": "testVNET",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
+    "scriptFolder": "scripts",
+    "scriptFileName": "testBandwidth.ps1",
+    "vhdStorageAccountName": "[concat(uniqueString(resourceGroup().id), 'storage')]",
+    "dataStorageAccountName": "[concat(uniqueString(resourceGroup().id), 'datastorage')]",
+    "frontEndNSGName": "[concat('webtestnsg-', uniqueString(resourceGroup().id))]",
+    "vmName": "testVM",
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
```

Comment:
can we use:
subnetref: resourceId('Microsoft.Network/virutualNetworks/subnets', variables('virtualNetworkName, variables('subnetName')